internal/gitutil/git_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitutil
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"sigs.k8s.io/krew/internal/testutil"
+)
+
+func assertRepoIsCloned(t *testing.T, repoPath string) {
+	t.Helper()
+
+	ok, err := IsGitCloned(repoPath)
```

Comment:
ideally test code shouldn't depend on functionality that's being tested. 

in this case, IsGitCloned should have a test.

so please consider actually using ioutil.ReadDir and verify the repoPath has >2 files, one of them being `.git/`.

Reply:
Sure. that sounds ok to me. I was doing it this way because i've specific tests to check if `IsGitCloned` worked

Reply:
Maybe you can also shell out to `git` to check if it is a repo. Then we don't even need to assume anything about the internals of git repositories.

Other than that, I think your approach has drawbacks as Ahmet pointed out, but it should be fine. After all, we can assume that `IsGitCloned` is pretty solid.